gbpservice/nfp/base_configurator/api/v1/controllers/controller.py

```diff
@@ -69,19 +86,36 @@ def get(self):
         Returns: Dictionary that contains Notification data
 
         """
-
+        global cache_ips
         global notifications
         try:
-            notification_data = jsonutils.dumps(notifications)
-            msg = ("NOTIFICATION_DATA sent to config_agent %s"
-                   % notification_data)
-            LOG.info(msg)
-            notifications = []
-            return notification_data
+            if not cache_ips:
+                notification_data = jsonutils.dumps(notifications)
+                msg = ("NOTIFICATION_DATA sent to config_agent %s"
+                       % notification_data)
+                LOG.info(msg)
+                notifications = []
+                return notification_data
+            else:
+                for ip in cache_ips:
+                    notification_response = requests.get(
+                        'http://'+str(ip)+':8080/v1/nfp/get_notifications')
+                    notification = jsonutils.loads(notification_response.text)
+                    notifications.extend(notification)
+                    cache_ips.remove(ip)
```

Comment:
This issue could be solved by adding ip to cache_ips after post request. I have made the changes on base_mode_vm branch.